src/mslice/plotting/plot_window/slice_plot.py

```diff
@@ -331,40 +332,43 @@ def update_canvas(self, cbar_range, cbar_log, x_range, y_range, title):
         self._canvas.draw()
 
     def _run_temp_dependent(self, slice_plotter_method, previous):
-        temp_value_raw = None
-        temp_value = None
         try:
             slice_plotter_method(self.ws_name)
-        except ValueError:  # sample temperature not yet set
-            try:
-                temp_value_raw, field = self.ask_sample_temperature_field(str(self.ws_name))
-            except RuntimeError:  # if cancel is clicked, go back to previous selection
-                self.set_intensity(previous)
-                return False
-            if field:
-                self._slice_plotter_presenter.add_sample_temperature_field(temp_value_raw)
-                temp_value = self._slice_plotter_presenter.update_sample_temperature_from_field(self.ws_name)
-            else:
-                temp_value = get_sample_temperature_from_string(temp_value_raw)
-                if temp_value is not None:
-                    try:
-                        temp_value = float(temp_value)
-                    except ValueError:
-                        temp_value = None
-            if temp_value is None or temp_value < 0:
-                self.plot_window.display_error("Invalid value entered for sample temperature. Enter a value in Kelvin \
-                                                or a sample log field.")
-                self.set_intensity(previous)
+        except ValueError:  # sample temperature not yet set, get it and reattempt method
+            if self._set_sample_temperature(previous):
+                slice_plotter_method(self.ws_name)
+            else:  # failed to get sample temperature
                 return False
-            else:
-                self.default_options['temp'] = temp_value
-                self.temp = temp_value
-                self._slice_plotter_presenter.set_sample_temperature(self.ws_name, temp_value)
-            slice_plotter_method(self.ws_name)
         return True
 
+    def _set_sample_temperature(self, previous):
+        try:
+            temp_value_raw, field = self.ask_sample_temperature_field(str(self.ws_name))
+            temperature_found = self._handle_temperature_input(temp_value_raw, field)
+        except RuntimeError:  # if cancel is clicked, go back to previous selection
+            temperature_found = False
+        if not temperature_found:
+            self.set_intensity(previous)
+        return temperature_found
+
+    def _handle_temperature_input(self, temp_value_raw, field):
```

Comment:
While we are doing a refactor in this area, could we add type hinting to these new functions?

Reply:
With pleasure